### PR TITLE
Clang tidy

### PR DESCRIPTION
**Pull request by bjorgve**
Clang-tidy modernize. This should kill some of the warning generated by clang. See if the changes makes sense. 

Particularly look at the use of default in Class headers to see if they make sense.

### CODE REVIEW

**Comment by robertodr** [comment]
Could you add documentation on how to run `clang-tidy`?

**Comment by robertodr** [comment]
Could you add documentation on how to run `clang-tidy`?

**Comment by bjorgve** [comment]
Sure!

File: src/core/QuadratureCache.cpp [hunk 1]
```diff
@@ -44,7 +44,7 @@ QuadratureCache::QuadratureCache() {
     this->intervals = 1;
 }
 
-QuadratureCache::~QuadratureCache() {}
+QuadratureCache::~QuadratureCache() = default;
```
Comment (stigrj) [inline src/core/QuadratureCache.cpp]: Can you move this to the header file?

File: src/functions/Gaussian.cpp [hunk 2]
```diff
@@ -58,7 +58,7 @@ Gaussian<D>::Gaussian(const std::array<double, D> &a, double c, const Coord<D> &
         , pos(r)
         , squareNorm(-1.0) {}
 
-template <int D> Gaussian<D>::~Gaussian() {}
+template <int D> Gaussian<D>::~Gaussian() = default;
```
Comment (stigrj) [inline src/functions/Gaussian.cpp]: Also header file

**Comment by bjorgve** [comment]
Yes, it uses the `compile_commands.json`

**Pull request merged**